management_app/views/faculty.py

```diff
@@ -8,9 +8,16 @@
 @faculty.route('/')
 @login_required
 def index():
+  # TODO: Get data from database
   faculties = [
     { 'name': 'Professor Mark',  'email': 'mark@uci.edu', 'role': 'Tenured Faculty', 'required_point': 3.5, 'prev_balance': 0.1250, 'estimated_balance': 0.2500 },
     { 'name': 'Professor 2',  'email': 'mark1@uci.edu', 'role': 'Pre Tenured Faculty 1 st', 'required_point': 2.5, 'prev_balance': 0.1000, 'estimated_balance': 0.1250 },
     { 'name': 'Professor 3',  'email': 'mark3@uci.edu', 'role': 'PoT Tenured Faculty', 'required_point': 6.5, 'prev_balance': 1.4375, 'estimated_balance': 0.9375 }
   ]
   return render_template('faculty/index.html', faculties=faculties)
+
+@faculty.route('/download_file/<filename>')
```

Comment:
If we are following REST then the endpoint should be named after a resource instead of an action.
```suggestion
@faculty.route('/templates/<filename>')
```
or
```suggestion
@faculty.route('/data-templates/<filename>')
```